### DISCUSSION TITLE
get a variables nodepath from the inspector

### DISCUSSION THREAD

**Discussion by oxixxixo** [godotengine/godot-proposals#7499] [Editor]
I'm working on a plugin that tracks things like position from a node and maps them onto another node as an animation. It would be nice to give the user the ability to choose any variable instead of hard coding for a nodes position or whatever. I'm assuming the ability is already there as the animationplayer tab can get a reference to these with the key icon. 

I'm wonder if it would be possible to add 'copy property nodepath' as an option when you right click, next to the 'copy property path' option. It could use the shortcut ctrl+alt+c and would return something like "Path2D/PathFollow2D/Sprite2D:texture:size" for example.

![image](https://github.com/godotengine/godot-proposals/assets/61514710/8ce93e3b-a65b-41fe-9bea-e50975e31384)

If someone's knows of some way to get the full variable nodepath in the editor in some other way I would also appreciate hearing about it, thanks

**Comment by theraot** [comment]
Yes, something like this can be implemented...

The problem is *from where* do you want the path, as when you copy the property path, it is not clear from where you will access it. And no, an absolute path won't work, because Godot doesn't know where the scene will be instantiated in the scene tree.

I think we could add an option to copy the path relative to root of the currently edited scene. *Which, I believe, is how "Copy Node Path" currently works.* Similarly, it might also be worth considering a modifier for dragging the property into an script.

Of course, the inspector must be on a `Node` for this to work.

---

In the case of the `AnimationPlayer` it has a `root_node` property that specifies from where it will get the property path.
